app/lib/features/deep_linking/utils.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:acter/features/deep_linking/types.dart';
+
+class UriParseError extends Error {}
+
+class SchemeNotSupported extends UriParseError {
+  final String scheme;
+
+  SchemeNotSupported({required this.scheme});
+}
+
+class ObjectNotSupported extends UriParseError {
+  final String objectType;
+
+  ObjectNotSupported({required this.objectType});
+}
+
+class ParsingFailed extends UriParseError {}
+
+UriParseResult parseUri(Uri uri) => switch (uri.scheme) {
```

Comment:
Highly wrapped code, but just check the unit tests to understand the formats and what it does ...